sc_demographics.R

```diff
@@ -0,0 +1,188 @@
+
+#####################################################
+# Convert Social Care Demographics
+# Author: Catherine Holland
+# Date: February 2022
+# Written on RStudio Server
+# Version of R - 3.6.1
+# Input -
+# Description -
+#####################################################
+
+
+## output ##
+output <- haven::read_sav(file = "/conf/hscdiip/SLF_Extracts/Social_care/sc_demographics_lookup_Dec_2021.zsav")
+
+
+
+## load packages ##
+library(readr)
+library(odbc)
+library(dplyr)
+library(stringr)
+library(dbplyr)
+library(phsmethods)
+
+
+
+## data ##
+
+######################################################
+# set-up conection to platform
+db_connection <- odbc::dbConnect(
+  odbc::odbc(),
+  dsn = "DVPROD",
+  uid = Sys.getenv("USER"),
+  pwd = rstudioapi::askForPassword("password")
+)
```

Comment:
Could you add this as a new function please:
```R
#' Open a connection to a PHS database
#'
#' @param dsn The Data Source Name passed on to `odbc::dbconnect`
#' the dsn must be setup first. e.g. SMRA or DVPROD
#' @param username The username to use for authentication,
#' if not supplied it first will check the environment variable
#' and finally ask the user for input.
#'
#' @return a connection to the specified dsn
#' @export
phs_db_connection <- function(dsn, username = Sys.getenv("USER")) {
  # Collect username from the environment
  username <- Sys.getenv("USER")

  # Check the username is not empty and take input if not
  if (is.na(username) | username == "") {
    username <- rstudioapi::showPrompt(
      title = "Username",
      message = "Username",
      default = ""
    )
  }

  # Create the connection
  password_text <- glue::glue("{dsn} password for user: {username}")
  db_connection <- odbc::dbConnect(
    odbc::odbc(),
    dsn = dsn,
    uid = username,
    pwd = rstudioapi::askForPassword(password_text)
  )

  return(db_connection)
}
```

Then this can be:
```suggestion
db_connection <- phs_db_connection(dsn = "DVPROD")
```

This will be used in all the other Social Care scripts.